src/librustc/query/mod.rs

```diff
@@ -861,8 +861,13 @@ rustc_queries! {
     }
 
     Codegen {
-        query collect_and_partition_mono_items(_: CrateNum)
-            -> (Arc<DefIdSet>, Arc<Vec<Arc<CodegenUnit<'tcx>>>>) {
+        query collect_and_partition_mono_items(_: CrateNum) -> (
+            Arc<DefIdSet>,
```

Comment:
This has collected quite the list of output arguments. I think it would be better to use a struct with named fields here